processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -192,19 +224,31 @@ protected Object getAggVal(Aggregator agg, int rowOffset, int aggPosition)
   @Override
   public float getMetricFloatValue(int rowOffset, int aggOffset)
   {
-    return aggList.get(rowOffset)[aggOffset].getFloat();
+    final Aggregator[] aggs = aggList.get(rowOffset);
+    if (null == aggs) {
+      return 0.0f;
+    }
+    return aggs[aggOffset].getFloat();
   }
 
   @Override
   public long getMetricLongValue(int rowOffset, int aggOffset)
   {
-    return aggList.get(rowOffset)[aggOffset].getLong();
+    final Aggregator[] aggs = aggList.get(rowOffset);
+    if (null == aggs) {
+      return 0l;
+    }
+    return aggs[aggOffset].getLong();
   }
 
   @Override
   public Object getMetricObjectValue(int rowOffset, int aggOffset)
   {
-    return aggList.get(rowOffset)[aggOffset].get();
+    final Aggregator[] aggs = aggList.get(rowOffset);
+    if (null == aggs) {
```

Comment:
I don't think we should ever allow this condition to happen. We never returned null in 0.6, and this could be devastating for complex metrics that always assume some initial non-null value (e.g. HLL).